modules/serverless-service-mesh-resources.adoc

```diff
@@ -0,0 +1,129 @@
+// Module included in the following assemblies:
+//
+// * serverless/networking/serverless-ossm-custom-domains.adoc
+
+[id="serverless-service-mesh-resources_{context}"]
+= Creating necessary {ProductShortName} resources
+
+.Procedure
+
+. Create an Istio gateway to accept traffic.
+.. Create a YAML file, and copy the following YAML into it:
++
+
+[source,yaml]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: default-gateway
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+----
+
+.. Apply the YAML file:
++
+
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Create an Istio `VirtualService` object to rewrite the host header.
+.. Create a YAML file, and copy the following YAML into it:
++
+
+[source,yaml]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: hello
+spec:
+  hosts:
+  - custom-ksvc-domain.example.com
+  gateways:
+  - default-gateway
+  http:
+  - rewrite:
+      authority: hello.default.svc
+    route:
+    - destination:
+        host: hello.default.svc
+        port:
+          number: 80
+----
+
+.. Apply the YAML file:
++
+
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Create an Istio `ServiceEntry` object. This is required for {ServerlessProductName} since Kourier is outside of the service mesh.
+.. Create a YAML file, and copy the following YAML into it:
++
+
+[source,yaml]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: hello.default.svc
+spec:
+  hosts:
+  - hello.default.svc
+  location: MESH_EXTERNAL
+  endpoints:
+  - address: kourier-internal.knative-serving-ingress.svc
+  ports:
+  - number: 80
+    name: http
+    protocol: HTTP
+  resolution: DNS
+----
+
+.. Apply the YAML file:
++
+
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Create an {product-title} route that points to the `VirtualService` object.
+.. Create a YAML file, and copy the following YAML into it:
++
+
+[source,yaml]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: hello
+spec:
+  host: custom-ksvc-domain.example.com
+  port:
+    targetPort: 8080
+  to:
+    kind: Service
+    name: istio-ingressgateway
+----
+
+.. Apply the YAML file:
++
+
+[source,terminal]
+----
+$ oc apply -n istio-system -f <filename>
```

Comment:
Maybe I'm just missing the final step of this procedure. That should be sending a `curl` request to the given service, maybe?

Reply:
Oh. I see it in the rendered document. It's in a different file. Cool.

Reply:
I think this is covered here? https://customdomain--ocpdocs.netlify.app/openshift-enterprise/latest/serverless/networking/serverless-ossm-custom-domains.html#serverless-access-custom-domain_serverless-ossm-custom-domains